src/dispatch/plugins/dispatch_slack/modals/incident/views.py

```diff
@@ -0,0 +1,324 @@
+
+    # switch from update to submit when we have a project
+    if project:
+        modal_template["callback_id"] = ReportIncidentCallbackId.submit_form
+        modal_template["blocks"] += [
+            incident_type_select_block(db_session=db_session, project_id=project.id),
+            incident_priority_select_block(db_session=db_session, project_id=project.id),
+            tag_multi_select_block(),
+        ]
+
+    return modal_template
+
+
+def update_participant(incident: Incident, participant: Participant = None):
+    """Builds all blocks required for updating the participant modal."""
+    modal_template = {
+        "type": "modal",
+        "title": {"type": "plain_text", "text": "Edit Participant"},
+        "blocks": [
+            {
+                "type": "context",
+                "elements": [
+                    {
+                        "type": "mrkdwn",
+                        "text": "Use this form to edit why a particpant was added to this incident.",
+                    }
+                ],
+            },
+            participants_select_block(incident=incident, initial_option=participant),
+        ],
+        "close": {"type": "plain_text", "text": "Cancel"},
+        "submit": {"type": "plain_text", "text": "Submit"},
+        "callback_id": UpdateParticipantCallbackId.update_view,
+        "private_metadata": json.dumps(
+            {"incident_id": str(incident.id), "channel_id": str(incident.conversation.channel_id)}
+        ),
+    }
+
+    # we need to show the reason if we're updating
+    if participant:
+        modal_template["blocks"].append(
+            {
+                "block_id": UpdateParticipantBlockId.reason_added,
+                "type": "input",
+                "element": {
+                    "type": "plain_text_input",
+                    "multiline": True,
+                    "initial_value": participant.added_reason or "",
+                    "action_id": UpdateParticipantBlockId.reason_added,
+                },
+                "label": {"type": "plain_text", "text": "Reason Added"},
+            }
+        )
+
+        modal_template["callback_id"] = UpdateParticipantCallbackId.submit_form
+
+    return modal_template
+
+
+def update_notifications_group(incident: Incident, db_session: SessionLocal):
+    """Builds all blocks required to update the membership of the notifications group."""
+    modal_template = {
+        "type": "modal",
+        "title": {"type": "plain_text", "text": "Update Group Membership"},
+        "blocks": [
+            {
+                "type": "context",
+                "elements": [
+                    {
+                        "type": "plain_text",
+                        "text": "Use this form to update the membership of the notifications group.",
+                    }
+                ],
+            },
+        ],
+        "close": {"type": "plain_text", "text": "Cancel"},
+        "submit": {"type": "plain_text", "text": "Update"},
```

Comment:
To be consistent with the other buttons
```suggestion
        "submit": {"type": "plain_text", "text": "Submit"},
```